libs/durability/near_full_data_storage.py

```diff
@@ -158,3 +171,71 @@ def perform_near_full_sys_operations(s3userinfo, workload_info: list, skipread:
                 return False, f"S3bench workload for failed for {each['obj_size']}." \
                               f" Please read log file {resp[1]}"
         return True, "S3bench workload successful"
+
+    @staticmethod
+    def delete_workload(workload_info_list: list, s3userinfo: dict, delete_percent: int):
+        """
+        Delete specified percent of buckets(with written data) from workload info list.
+        :param workload_info_list: Workload info list returned after write operation
+        :param s3userinfo: User info for IAM user
+        :param delete_percent: Percentage for deletion
+        :return : Tuple(boolean,str)
+        """
+        num_buckets_delete = int(delete_percent * len(workload_info_list) / 100)
+        delete_list = []
+        LOGGER.info("Delete %s random buckets.", num_buckets_delete)
+
+        for _ in range(num_buckets_delete):
+            bucket_info = workload_info_list[
+                random.SystemRandom().randint(1, len(workload_info_list) - 1)]
+            delete_list.append(bucket_info)
+            workload_info_list.remove(bucket_info)
+        LOGGER.info("Deleting buckets : %s", delete_list)
+        resp = NearFullStorage.perform_operations_on_pre_written_data(
+            s3userinfo=s3userinfo,
+            workload_info=delete_list,
+            skipread=False,
+            validate=True,
+            skipcleanup=False)
+
+        if resp[0]:
+            deleted_buckets = [each['bucket'] for each in delete_list]
```

Comment:
why is this needed?

Reply:
As new bucket creation/deletion is not allowed, we are reusing the buckets post deletion . 
hence retrieving bucket names from workload info for reuse.